### PR TITLE
HPCC-17271 Add options to layout translation

### PR DESCRIPTION
**Pull request by richardkchapman**
Added new "payload" option as well as "true" or "false"

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

## Testing:
Regression suite running as I type. I originally had it defaulting to "payload only" which meant the regression suite failed (and tested that the error was being reported properly) before changing my mind and leaving default as it was for back-compatibility.

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Plese review

File: common/workunit/pkgimpl.hpp [hunk 1]
```diff
@@ -118,13 +118,20 @@ class WORKUNIT_API CPackageNode : implements IHpccPackage, public CInterface
...
-    virtual bool getSysFieldTranslationEnabled() const {return false;}
-    virtual bool getEnableFieldTranslation() const
+    virtual IRecordLayoutTranslator::Mode getSysFieldTranslationEnabled() const { return IRecordLayoutTranslator::NoTranslation; }
+    virtual IRecordLayoutTranslator::Mode getEnableFieldTranslation() const
     {
         const char *val = queryEnv("control:enableFieldTranslation");
         if (!val) val = queryEnv("enableFieldTranslation"); // Backward compatibility
         if (val)
-            return strToBool(val);
+        {
```
Comment (ghalliday) [inline common/workunit/pkgimpl.hpp]: potential for this to go in a shared function - since it is duplicated 3 times.

**Review by ghalliday — COMMENTED** [review]
@richardkchapman the changes looks good.  I would merge as is, but would be better if there was a test case for the keyed only - especially if that was included in the regression suite (both success and failure).

File: common/workunit/pkgimpl.hpp (hunk 1 above)
Comment (richardkchapman) [inline common/workunit/pkgimpl.hpp]: I wondered, but decided it wan't worth the bother.

**Pull request merged**